cmd/service_linux.go

```diff
@@ -0,0 +1,335 @@
+	Use:               "uninstall",
+	Short:             "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	Long:              "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	SuggestFor:        []string{"uninst", "uninstal"},
+	Example:           "cloudfuse service uninstall --mount-path=<path/to/mount/path>",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// get absolute path of provided relative mount path
+		if strings.Contains(serviceName, ".") || strings.Contains(serviceName, "..") {
+			dir, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("error: [%s]", err.Error())
+			}
+			mountPath = common.JoinUnixFilepath(dir, serviceName)
+
+		}
+
+		// get service file name and service file path
+		folderList := strings.Split(serviceName, "/")
+		serviceName = folderList[len(folderList)-1] + ".service"
+		servicePath := "/etc/systemd/system/" + serviceName
+
+		// delete service file
+		removeFileCmd := exec.Command("sudo", "rm", servicePath)
+		err := removeFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to delete "+serviceName+" file from /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		// reload daemon
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func newServiceFile(mountPath string, configPath string, serviceUser string) (string, error) {
+	serviceTemplate := ` [Unit]
+	Description=Cloudfuse is an open source project developed to provide a virtual filesystem backed by S3 or Azure storage.
+	After=network-online.target
+	Requires=network-online.target
+
+	[Service]
+	# User service will run as.
+	User={{.ServiceUser}}
+	# Path to the location Cloudfuse will mount to. Note this folder must currently exist.
+	Environment=MountingPoint={{.MountPath}}
+	# Path to the configuration file.
+	Environment=ConfigFile={{.ConfigFile}}
+
+	# Under the hood
+	Type=forking
+	ExecStart=/usr/bin/cloudfuse mount ${MountingPoint} --config-file=${ConfigFile} -o allow_other
+	ExecStop=/usr/bin/fusermount -u ${MountingPoint} -z
+
+	[Install]
+	WantedBy=multi-user.target
+	`
+
+	config := serviceOptions{
+		ConfigFile:  configPath,
+		MountPath:   mountPath,
+		ServiceUser: serviceUser,
+	}
+
+	tmpl, err := template.New("service").Parse(serviceTemplate)
+	if err != nil {
+		fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	folderList := strings.Split(mountPath, "/")
+	serviceName := folderList[len(folderList)-1] + ".service"
+	newFile, err := os.Create("/etc/systemd/system/" + serviceName)
+	if err != nil {
+		return "", fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
```

Comment:
Could you overwrite it, in case they want to use the install command to update an existing service?
Or would you rather they uninstall first?